common/wuanalysis/anawu.cpp

```diff
@@ -0,0 +1,391 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "jliball.hpp"
+
+#include "workunit.hpp"
+#include "anacommon.hpp"
+#include "anarule.hpp"
+#include "anawu.hpp"
+
+class WuScope;
+class WorkunitAnalyser;
+
+class WuScopeHashTable : public SuperHashTableOf<WuScope, const char>
+{
+public:
+    ~WuScopeHashTable() { _releaseAll(); }
+
+    void addNew(WuScope * newWuScope) { SuperHashTableOf::addNew(newWuScope);}
```

Comment:
does this do anything?  Left over from debugging?

Reply:
This is used in WuScope::select to add a new scope (if it's not already in the hash table)